src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -482,19 +480,20 @@ protected void setupChild(Configuration childConf)
         if (child instanceof FileSetCheck) {
             final FileSetCheck fsc = (FileSetCheck) child;
             fsc.init();
-            addFileSetCheck(fsc);
+            fsc.setMessageDispatcher(this);
+            fileSetChecks.add(fsc);
```

Comment:
why we try to not use addXXXX methods ? in this specific case it is not single statement wrapper.

Reply:
this most likely an answer - https://github.com/checkstyle/checkstyle/pull/12511/files#r1046651114

Reply:
does it make sense do make private method something line `addFileSetCheckInternal` to keep two statement together.

Reply:
if we do this update only for future update, lets not do this now, it is very small, it will looks more natural in next PR when addListenter start to have new code that we do not want to apply each time.

Reply:
@rnveach , please reply.

Reply:
this method was used only here
![image](https://github.com/checkstyle/checkstyle/assets/812984/dac3383c-906a-4efb-b549-8c4c1282905f)

it is public, so theoretically can be used by plugins, but we are kind deprecating it ? we need to figureout how much it will affect plugins/extensions.